### PR TITLE
refactor, moving functions into attr and htmlfontelemend fixes #9639 …

### LINKED ISSUE #9639: Move parse_nonzero_length and parse_legacy_color from util::str to style::attr
Files:
* components/util/str.rs
* components/style/attr.rs

### PR DESCRIPTION
**Pull request by jasonwilliams**
…#9638

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9647)

### CODE REVIEW

**Comment by nox** [comment]
-S-awaiting-review +S-needs-code-changes 

---

Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[components/script/dom/htmlfontelement.rs, line 18 \[r1\]](https://reviewable.io:443/reviews/servo/servo/9647#-KA_vT5u7HbAQEvIQAjs:-KA_vT5u7HbAQEvIQAjt:-1553075598) ([raw file](https://github.com/servo/servo/blob/46b662b49cfc25dccde152111bfc1c6f0a3770e5/components/script/dom/htmlfontelement.rs#L18)):*
I meant between DOMString and read_numbers.

---

*[components/style/attr.rs, line 12 \[r1\]](https://reviewable.io:443/reviews/servo/servo/9647#-KA_vWgzEs4i87flv9-S:-KA_vWgzEs4i87flv9-T:1028044108) ([raw file](https://github.com/servo/servo/blob/46b662b49cfc25dccde152111bfc1c6f0a3770e5/components/style/attr.rs#L12)):*
Nit: WHITESPACE after LengthOrPercentageOrAuto.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9647#-:-KA_vd2J720DpeiTBZVO:1458266033)*

**Comment by nox** [comment]
@bors-servo r+

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9647#-:-KAa-1nBgbznfv3xSkls:1236647940)*

**Comment by jdm** [comment]
```
./components/util/str.rs:369: no newline at EOF
```

File: components/util/str.rs [hunk 1]
```diff
@@ -564,4 +366,4 @@ pub fn search_index(index: usize, indices: CharIndices) -> isize {
         character_count += 1
     }
     character_count
-}
+}
```
Comment (KiChjang) [inline components/util/str.rs]: Tidy isn't happy about this change.

**Comment by nox** [comment]
@bors-servo r+

-S-fails-tidy

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9647#-:-KAb9xaKLUo10ozUNtQi:-2046478299)*

**Pull request merged**